### ISSUE TITLE
sample: balanced let block

### ISSUE THREAD

**Issue by zimbatm** [nix-community/nixpkgs-fmt#65]
It would better if the body was balanced with the set entries when the let block is not on the root of the file:

## Input
```nix
{
  myAttr = foo:
    let
 hello = 3; in hello;
}
```

## Output
```nix
{
  myAttr = foo:
    let
      hello = 3;
    in
    hello;
}

```

## Desired output
```nix
{
  myAttr = foo:
    let
      hello = 3;
    in
      hello;
}

```

**Comment by aij** [comment]
I'm somewhat partial to the more compact form, which avoids extra indentation for new scopes introduced by let bindings. (This may be due to having experience with OCaml and SML though, where it's really common to use smaller scopes.)

Eg:

```nix
{
  myAttr = foo:
    let hello = 3; in
    hello;
}
```

There is at least some precedence for this in `nixpkgs`.  I'm finding about 374 examples via the regex `let .* = .*; in`

**Comment by matklad** [comment]
I also feel that for "header" constructs like

```
with expr;
body

assert cond;
body

let
  bindings
in
body
```

it makes sense *not* to indent the body. I feel that way because these construct work more or less like just `body`. 

That is, unlike with a lambda, body expression is just evaluated.

**Comment by matklad** [comment]
was closed by #79
